notebooks/ch-bdqc/delegated-computation.ipynb

```diff
@@ -0,0 +1,472 @@
+    "    qc.cz(i,i+1)\n",
+    "\n",
+    "####END Phase 1: preparation phase\n",
+    "\n",
+    "\n",
+    "##########Measurement\n",
+    "def x_measurement(qbit:int):\n",
+    "    # the function meausures the MSB as the bit on the left (for consistency with the litterature, as opposed to usual order in qiskit)\n",
+    "    # Measurement is done in the X basis, i.e Hadamard before measurement\n",
+    "\n",
+    "    qc.h(qbit)\n",
+    "    qc.measure(q[qbit],c[nb_reg-1-qbit])\n",
+    "\n",
+    "\n",
+    "####Phase 2: interaction and sequentiel measurement\n",
+    "qb=0\n",
+    "delta=  computation_angle[qb] - (A_hidden_phase[qb] *np.pi/4)  -  (random_r[qb]  *np.pi)\n",
+    "qc.p(delta ,qb)\n",
+    "x_measurement(qb)\n",
+    "\n",
+    "#qubit 0 will give X correction if =1 to qubit 1\n",
+    "with qc.if_test((c[nb_reg-1-(0)], 1)):\n",
+    "    qc.x(1)        \n",
+    "\n",
+    "\n",
+    "\n",
+    "#qubit 1 will give X correction if =1 to qubit 2\n",
+    "with qc.if_test((c[nb_reg-1-(1)], 1)):\n",
+    "    qc.x(2)\n",
+    "\n",
+    "#qubit 0 will give Z correction if =1 to qubit 2\n",
+    "with qc.if_test((c[nb_reg-1-(0)], 1)):\n",
+    "    qc.z(2)        \n",
+    "\n",
+    "\n",
+    "qb=2\n",
+    "delta=  computation_angle[qb] - (A_hidden_phase[qb] *np.pi/4)  -  (random_r[qb]  *np.pi)\n",
+    "qc.p(delta ,qb)\n",
+    "x_measurement(qb)\n",
+    "\n",
+    "qc.draw(output=\"mpl\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'000': 5069, '100': 4931}\n"
+     ]
+    }
+   ],
+   "source": [
+    "### Running the simulator to obtain measurement outcome\n",
+    "simulator = Aer.get_backend('aer_simulator_statevector') #qasm simulator\n",
+    "transpiled_circuit = transpile(qc, simulator)\n",
+    "job = execute(qc,simulator,shots = 10000)\n",
+    "result = job.result()\n",
+    "counts = result.get_counts(qc)\n",
+    "print(counts)\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you may have noticed the above implementation works in the sense that we teleport a `0` in the end. But, what we get in the end is not the same 4 outcomes we saw in the MBQC tutorial even though it is supposed to be the same computation, just with blindness. So, this should be a sign that something might be wrong.\n",
+    "\n",
+    "Also, there is a major flaw, we are doing the MBQC correction separetly from the unblinding and the random $r_i\\pi$ rotation. This means that we are leaking to the server the real value of the previous qubit based on the corrected outcome and the choosen correction. What we have to do in order to perform step 5 correctly is to adapt the measurement basis by including the MBQC correction in it, i.e modify the computation value $\\phi_{i+1}$ instead of doing X and Z operations separately.\n",
```

Comment:
```suggestion
    "Also, there is a major flaw, we are doing the MBQC correction separately from the unblinding and the random $r_i\\pi$ rotation. This means that we are leaking to the server the real value of the previous qubit based on the corrected outcome and the chosen correction. What we have to do in order to perform step 5 correctly is to adapt the measurement basis by including the MBQC correction in it, i.e. modify the computation value $\\phi_{i+1}$ instead of doing X and Z operations separately.\n",
```